src/hooks/useWindowDimensions.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+const useWindowDimensions = () => {
+  const [windowDimensions, setWindowDimensions] = useState({
+    width: undefined,
+    height: undefined,
+  });
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions({
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    }
+    handleResize();
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  return windowDimensions;
+};
+
+export default useWindowDimensions;
```

Comment:
When opening the page for the first time, sometimes is takes longer than expected to check for the availability of the `window` API. Before it is found in the browser `width` and `height` cannot access `window.innerWidth` and `window.innerHeight` which throws an error. By setting it to undefined first and handling the Resize in line 15 it makes sure to avoid this error by setting width and height to `undefined` first and then reset using the `window` API when ready.